tests/functional_tests/test_query_builder.py

```diff
@@ -4,6 +4,95 @@
 import io
 import pandas as pd
 from datenguidepy import Field, Query
+from datenguidepy.query_execution import FieldMetaDict, TypeMetaData
+
+
+@pytest.fixture
+def mocked_meta_data(monkeypatch):
+    def mock_enum_placeholder(self):
+        return "MOCKED ENUM VALUES"
+
+    def mock_get_arguments(self):
+        mocked_args = {
+            "year": ("LIST", None, "SCALAR", "Int"),
+            "statistics": ("LIST", None, "ENUM", "BEV001Statistics"),
+            "ALTMT1": ("LIST", None, "ENUM", "ALTMT1"),
+        }
+        return mocked_args
+
+    def mock_get_description(self, *args):
+        return "Lebend Geborene"
+
+    def mock_field_meta_data(self):
+
+        type_kind = "kind='OBJECT'"
+        field_meta = {
+            "id": FieldMetaDict(
+                {
+                    "name": "id",
+                    "type": {
+                        "ofType": None,
+                        "kind": "SCALAR",
+                        "name": "String",
+                        "description": "The `String` scalar...",
+                    },
+                    "description": "Interne eindeutige ID",
+                    "args": [],
+                }
+            ),
+            "year": FieldMetaDict(
+                {
+                    "name": "year",
+                    "type": {
+                        "ofType": None,
+                        "kind": "SCALAR",
+                        "name": "Int",
+                        "description": "The `Int` scalar...",
+                    },
+                    "description": "Jahr des Stichtages",
+                    "args": [],
+                }
+            ),
+            "value": FieldMetaDict(
+                {
+                    "name": "value",
+                    "type": {
+                        "ofType": None,
+                        "kind": "SCALAR",
+                        "name": "Float",
+                        "description": "The `Float` scalar...",
+                    },
+                    "description": "Wert",
+                    "args": [],
+                }
+            ),
+        }
+        enum_vals = None
+        return Field._no_none_values(
```

Comment:
Here some of the functionality of `Field.fields_info` is reproduced. It would direct to mock the `_graphql_schema_meta_data_provider.type_info` method instead of the `Field.fields_info` method. then the formatting function would be called automatically and would be tested in its proper place. Right now it is implicitly tested as part of the mock function.
Conceptually when testing the field class it would also be more normal to mock the external dependencies, namely the meta data providers instead of the classes methods themselves. Although one might argue that the Field class is an external dependency of the Query class and is therefore mocked.
As a side note most private formatted functions appear to be static and would ideally be marked as such.

Reply:
I first tryed to mock `_graphql_schema_meta_data_provider.type_info` though the problem is, that for the "return_type" this function is called multiple times. - is there a better way (e.g. "only mock when input xy") than writing an "if else" for every time the function is used in this test?

Reply:
Ok I see the issue, I don't think there is a quick solution in terms of implementation. But I wonder in terms of test design whether it is necessary to have tests that mock part of an external dependency as opposed to having tests that mock it completely or not at all, which I find conceptually easier to grasp. Do you think we would loose some important test coverage in the latter case?